Util/Inc/bsp_config.h

```diff
@@ -0,0 +1,210 @@
+#ifndef BSP_CONFIG_H
+#define BSP_CONFIG_H
+
+#include <stdint.h>
+
+#define STR(x) #x // see below
+#define ENQUOTE(x) STR(x) // preprocessor black magic to print string equivalent of #defines
```

Comment:
I don't think you need this, you should be able to just do blah#blah to make blahblah

Reply:
We actually do need this nesting so that the #x expands the macro we try to print. See following:
![image](https://github.com/lhr-solar/Embedded-Sharepoint/assets/107018918/92e0392d-450e-4871-9b72-d62481c4fbdc)
This is without the nested macro expansion. With the ENQUOTE / STR combo, the preprocessor will actually expand out the values of the #defines (I.e., PA5_ALTERNATE_FUNCTION expands out to, say, GPIO, and that's what's printed.)
![image](https://github.com/lhr-solar/Embedded-Sharepoint/assets/107018918/e6809ab4-36b0-46ed-8f2b-7895c06222b1)